linkerd/failure-accrual/src/test/scala/io/buoyant/linkerd/failureAccrual/ConfigTest.scala

```diff
@@ -0,0 +1,119 @@
+package io.buoyant.linkerd.failureAccrual
+
+import com.twitter.finagle.service.Backoff
+import com.twitter.util.Duration
+import io.buoyant.config.Parser
+import io.buoyant.linkerd.{ConstantBackoffConfig, FailureAccrualConfig, JitteredBackoffConfig}
+import io.buoyant.test.FunSuite
+import org.scalatest.{Matchers, OptionValues, OutcomeOf}
+import com.twitter.conversions.time._
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.prop.{PropertyChecks, TableDrivenPropertyChecks}
+import org.scalacheck.Gen
+import org.junit.runner.RunWith
+
+/**
+ * Created by eliza on 6/27/17.
+ */
+@RunWith(classOf[JUnitRunner])
+class ConfigTest extends FunSuite
+  with Matchers
+  with OptionValues
+  with PropertyChecks
+  with OutcomeOf {
+
+  def parse(yaml: String): FailureAccrualConfig = {
+    val mapper = Parser.objectMapper(yaml, Seq(Seq(
+      new ConsecutiveFailuresInitializer,
+      new NoneInitializer,
+      new SuccessRateInitializer,
+      new SuccessRateWindowedInitializer
+    )))
+    mapper.readValue[FailureAccrualConfig](yaml)
+  }
+
+  val kinds =
+    Table(
```

Comment:
For my education: what's the advantage of table based checks rather than iterating over a collection and asserting?